### PR TITLE
Feature/divide integration tests

### PR DESCRIPTION
**Pull request by agduncan94**
* Divides integration tests into 5 parts (previously 3)
1. Client IT
2. Workflow IT
3. Tool IT
4. Toil IT
5. Other IT (Catch all)
* Previously we also ran a mvn clean install in the before_install section to check for dependencies and check swagger. This has been moved to the check-swagger script and is only run on Unit Tests.

#1457

### CODE REVIEW

**Comment by denis-yuen** [comment]
Note to self: coverage is actually 75.53% in coveralls

File: dockstore-common/src/main/java/io/dockstore/common/ToolTest.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package io.dockstore.common;
+
+public class ToolTest {
```
Comment (denis-yuen) [inline dockstore-common/src/main/java/io/dockstore/common/ToolTest.java]: Need a class comment on this

File: dockstore-common/src/main/java/io/dockstore/common/WorkflowTest.java [hunk 2]
```diff
@@ -0,0 +1,4 @@
+package io.dockstore.common;
+
+public class WorkflowTest {
```
Comment (denis-yuen) [inline dockstore-common/src/main/java/io/dockstore/common/WorkflowTest.java]: Same here (i.e. this is intended for workflows, Maven uses this, etc.)

**Comment by agduncan94** [comment]
I did not add the parallel builds to the travis.yml as described here.
I did ask about using parallel builds with a public repo at https://github.com/lemurheavy/coveralls-public/issues/1139. I'll add a comment about this within the code.

File: pom.xml [hunk 3]
```diff
@@ -1145,6 +1145,35 @@
...
+            <properties>
+                <skipTests>true</skipTests>
+                <skipITs>false</skipITs>
+                <groups>io.dockstore.common.ToolTest</groups>
+                <excludeGroups>io.dockstore.common.SlowTest,io.dockstore.common.RegressionTest, io.dockstore.common.ToilOnlyTest</excludeGroups>
+            </properties>
+        </profile>
+        <profile>
+            <id>other-integration-tests</id>
+            <!-- Only run integration tests not related to only tools or workflows. This is a catch all. -->
```
Comment (denis-yuen) [inline pom.xml]: I like the catch-all idea. It helps prevent tests from slipping through the cracks which was my concern.

**Comment by denis-yuen** [comment]
FYI, failing due to quay.io swagger document being blocked. 
Probably due to their outage. We may need to switch the location (save a copy) although the bigger issue is if they've blocked any actual calls outright.

**Pull request merged**